tests/monitoring/monitoring.go

```diff
@@ -637,8 +642,110 @@ var _ = Describe("[Serial][sig-monitoring]Prometheus Alerts", Serial, decorators
 			libwait.WaitForVirtualMachineToDisappearWithTimeout(vmi, 240)
 		})
 	})
+
+	Context("System Alerts", func() {
+		AfterEach(func() {
+			waitUntilAlertDoesNotExist("KubeVirtNoAvailableNodesToRunVMs")
+		})
+
+		disableVirtHandler := func() *v1.KubeVirt {
+			originalKv := util.GetCurrentKv(virtClient)
+			kv, err := virtClient.KubeVirt(originalKv.Namespace).Get(originalKv.Name, &metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			kv.Spec.CustomizeComponents = v1.CustomizeComponents{
+				Patches: []v1.CustomizeComponentsPatch{
+					{
+						ResourceName: virtHandler.deploymentName,
+						ResourceType: "DaemonSet",
+						Patch:        `{"spec":{"template":{"spec":{"nodeSelector":{"kubernetes.io/hostname":"does-not-exist"}}}}}`,
+						Type:         v1.StrategicMergePatchType,
+					},
+				},
+			}
+
+			kv, err = virtClient.KubeVirt(originalKv.Namespace).Update(kv)
+			Expect(err).ToNot(HaveOccurred())
+
+			Eventually(func() string {
+				vh, err := virtClient.AppsV1().DaemonSets(originalKv.Namespace).Get(context.Background(), virtHandler.deploymentName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				return vh.Spec.Template.Spec.NodeSelector["kubernetes.io/hostname"]
+			}, 90*time.Second, 5*time.Second).Should(Equal("does-not-exist"))
+
+			return kv
+		}
+
+		setEmulation := func(status bool) *v1.KubeVirt {
+			originalKv := util.GetCurrentKv(virtClient)
+			kv, err := virtClient.KubeVirt(originalKv.Namespace).Get(originalKv.Name, &metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			kv.Spec.Configuration.DeveloperConfiguration.UseEmulation = status
+
+			kv, err = virtClient.KubeVirt(originalKv.Namespace).Update(kv)
```

Comment:
Again this can become flaky. Use patch or `Eventually`

Reply:
changed